### ISSUE TITLE
PLATFORM=broadcom fails arista "ReloadCauseManagerTest" first time

### ISSUE THREAD

**Issue by chrispsommers** [Azure/sonic-buildimage#7800]
#### Description
When building broadcom PLATFORM, arista test fails. If you re-rerun `make all` again the build completes w/o error

#### Steps to reproduce the issue:
1. `make configure PLATFORM=broadcom`
2. `make all`

#### Describe the results you received:
```
...
============================= test session starts ==============================
platform linux2 -- Python 2.7.16, pytest-3.10.1, py-1.7.0, pluggy-0.8.0
rootdir: /sonic/platform/broadcom/sonic-platform-modules-arista, inifile: pytest.ini
plugins: cov-2.6.0
collected 112 items

arista/cli/tests/cli_platforms.py ...............                        [ 13%]
arista/components/denali/tests/chassis.py ...                            [ 16%]
arista/components/psu/tests/models.py .                                  [ 16%]
arista/core/tests/cause.py ......F.....                                  [ 27%]
arista/core/tests/components.py .                                        [ 28%]
arista/core/tests/fabrics.py ..                                          [ 30%]
arista/core/tests/inventory.py .......                                   [ 36%]
arista/core/tests/linecards.py ...                                       [ 39%]
arista/core/tests/platform_register.py ..s                               [ 41%]
arista/core/tests/platforms.py ............                              [ 52%]
arista/core/tests/prefdl.py .........                                    [ 60%]
arista/core/tests/psu.py ....                                            [ 64%]
arista/core/tests/registers.py ..........                                [ 73%]
arista/core/tests/thermal.py .......                                     [ 79%]
arista/core/tests/utils.py ..................                            [ 95%]
arista/core/tests/xcvr.py .                                              [ 96%]
arista/descs/tests/types.py ....                                         [100%]

=================================== FAILURES ===================================
________________ ReloadCauseManagerTest.testReloadCauseManager _________________

self = <arista.core.tests.cause.ReloadCauseManagerTest testMethod=testReloadCauseManager>

    def testReloadCauseManager(self):
       inv = self._getReloadCauseInventory()
>      self.rcm.readCauses(inv)

arista/core/tests/cause.py:129: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
arista/core/cause.py:212: in readCauses
    report = ReloadCauseReport(date=date or bootDatetime())
arista/libs/procfs.py:12: in bootDatetime
    return datetime.datetime.now() - datetime.timedelta(seconds=uptime()[0])
arista/libs/procfs.py:8: in uptime
    return tuple(float(v) for v in readFileContent(path).rstrip().split())
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

path = '/proc/uptime'

    def readFileContent(path):
>      with open(path) as f:
E      IOError: [Errno 2] No such file or directory: '/proc/uptime'

arista/libs/fs.py:5: IOError
----------------------------- Captured stdout call -----------------------------
DEBUG: No prior reboot cause information from /tmp/unittest-arista-rcm-XGqNww.json
=============== 1 failed, 110 passed, 1 skipped in 24.86 seconds ===============
make[3]: *** [Makefile:158: test-py2] Error 1
make[3]: Leaving directory '/sonic/platform/broadcom/sonic-platform-modules-arista'
make[2]: *** [debian/rules:56: override_dh_auto_test] Error 2
make[2]: Leaving directory '/sonic/platform/broadcom/sonic-platform-modules-arista'
make[1]: *** [debian/rules:14: build] Error 2
make[1]: Leaving directory '/sonic/platform/broadcom/sonic-platform-modules-arista'
dpkg-buildpackage: error: debian/rules build subprocess returned exit status 2
[  FAIL LOG END  ] [ target/debs/buster/sonic-platform-arista_1.0_amd64.deb ]
umount: /sonic/dpkg/tmp.WrgPg5AlLD: umount failed: No such file or directory.
make: *** [slave.mk:457: target/debs/buster/sonic-platform-arista_1.0_amd64.deb] Error 1
DEPRECATION: Python 2.7 reached the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 is no longer maintained. pip 21.0 will drop support for Python 2.7 in January 2021. More details about Python 2 support in pip can be found at https://pip.pypa.io/en/latest/development/release-process/#python-2-support pip 21.0 will remove support for this functionality.
Makefile.work:291: recipe for target 'all' failed
make[1]: *** [all] Error 2
make[1]: Leaving directory '/home/chris/sonic/sonic-buildimage'
Makefile:7: recipe for target 'all' failed
make: *** [all] Error 2
```
#### Describe the results you expected:
Build w/o any errors

#### Git Commit #:
```
git log
commit b30cf44b93b61e7c240c7c507786cc8355ac01f8
```

**Comment by zhangyanzhao** [comment]
@Staphylo please take a look

**Comment by Staphylo** [comment]
@zhangyanzhao I'll have a look, it seems to be an issue where the procfs is restricted for the build user.
I'll make the existance of /proc/uptime optional during the test phase.

**Comment by chrispsommers** [comment]
Thanks for looking. Note, I'm building inside an Ubuntu 16.04 VM running in VirtualBox on an Ubuntu 20.04 host. Perhaps this has an impact. I think it's a worthwhile use-case.

**Comment by Staphylo** [comment]
Thanks for the additional information Chris.
We do not build in such environment and it appears that Azure builds do not either which is why there is a gap.
The issue likely originate from the Ubuntu 16.04 VM which must be using a VM specific kernel which removes a bunch of kernel options or alternatively a locked down user environment. I'd also suggest upgrading your build environment to a more recent version of Ubuntu since 16.04 is showing its age and is already `End of Standard Support`.
We'll do our best to ensure that our unit testing does not try to read `/proc` entries in the future.

The fix is pretty much ready but it will take a few days until we open a PR and it merges.

**Comment by Staphylo** [comment]
I believe this issue should be closed. The fix merged with #8047 2 weeks ago.
I'm planning on making a separate PR to fix the issue on 202012 as well.

**Comment by Staphylo** [comment]
Please close this issue.
